File: 0226-invert-binary-tree/0226-invert-binary-tree.py
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def invert(original, result):
            if original.left:
                result.right = TreeNode(val=original.left.val)
                invert(original.left, result.right)
            if original.right:
                result.left = TreeNode(val=original.right.val)
                invert(original.right, result.left)


        if not root: return None

        answer = TreeNode(root.val)
        invert(root, answer)
        return answer

File: 0226-invert-binary-tree/0226-invert-binary-tree.java
class Solution {
    private void invert(TreeNode original, TreeNode result) {
        if (original.left != null) {
            result.right = new TreeNode(original.left.val);
            invert(original.left, result.right);
        }
        if (original.right != null) {
            result.left = new TreeNode(original.right.val);
            invert(original.right, result.left);
        }
    }

    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return root;
        }

        TreeNode result = new TreeNode(root.val);
        invert(root, result);
        return result;
    }
}